services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainController.java

```diff
@@ -94,6 +96,7 @@
   private volatile SyncService syncService;
   private volatile AttestationManager attestationManager;
   private volatile ValidatorCoordinator validatorCoordinator;
+  private CombinedChainDataClient combinedChainDataClient;
```

Comment:
There is no `volatile` keyword on the `combinedChainDataClient`, which is inconsistent with the other fields, is that intentional?

Reply:
Good spot.  Kind of annoying that we need volatile there and would be good to refactor so everything is created by a builder and injected to the constructor, but would have to separate construction from starting properly across all those classes first.